src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -174,15 +174,15 @@
     public static final String MSG_KEY = "forbid.escaped.unicode.char";
 
     /** Regular expression for Unicode chars. */
-    private static final Pattern UNICODE_REGEXP = Pattern.compile("\\\\u[a-fA-F0-9]{4}");
+    private static final Pattern UNICODE_REGEXP = Pattern.compile("\\\\u+[a-fA-F0-9]{4}");
 
     /**
      * Regular expression Unicode control characters.
      *
      * @see <a href="https://en.wiktionary.org/wiki/Appendix:Control_characters">
      *     Appendix:Control characters</a>
      */
-    private static final Pattern UNICODE_CONTROL = Pattern.compile("\\\\[uU]"
```

Comment:
`uU`
No need for U here because `\U` is invalid escape sequence anyway

Reply:
Please add in test input a case with upper U to make sure it wit not appear again in our code

Reply:
> Please add in test input a case with upper U to make sure it wit not appear again in our code

@romani what kind of input you mean? `\U` is not a valid escape sequence, code will not compile.